native_lib/DDGCmdline.c

```diff
@@ -0,0 +1,134 @@
+                           crypto_pwhash_MEMLIMIT_INTERACTIVE,
+                           crypto_pwhash_ALG_DEFAULT)) {
+
+        return DDGSYNCCRYPTO_CREATE_PRIMARY_KEY_FAILED;
+    }
+
+    if (0 != crypto_kdf_derive_from_key(passwordHash,
+                                        DDGSYNCCRYPTO_HASH_SIZE,
+                                        DDGSyncCryptoPasswordHashSubkey,
+                                        DDGSYNC_PASSWORD_HASH_CONTEXT,
+                                        primaryKey)) {
+
+        return DDGSYNCCRYPTO_CREATE_PASSWORD_HASH_FAILED;
+    }
+
+    if (0 != crypto_kdf_derive_from_key(stretchedPrimaryKey,
+                                        DDGSYNCCRYPTO_STRETCHED_PRIMARY_KEY_SIZE,
+                                        DDGSyncCryptoStretchedPrimaryKeySubkey,
+                                        DDGSYNC_STRETCHED_PRIMARY_KEY_CONTEXT,
+                                        primaryKey)) {
+
+        return DDGSYNCCRYPTO_CREATE_STRETCHED_PRIMARY_KEY_FAILED;
+    }
+
+    randombytes_buf(nonceBytes, crypto_secretbox_NONCEBYTES);
+
+    if (0 != crypto_secretbox_easy(protectedSecretKey,
+                                   secretKey,
+                                   DDGSYNCCRYPTO_SECRET_KEY_SIZE,
+                                   nonceBytes,
+                                   stretchedPrimaryKey)) {
+        return DDGSYNCCRYPTO_CREATE_PROTECTED_SECRET_KEY_FAILED;
+    }
+
+    memcpy(&protectedSecretKey[crypto_secretbox_MACBYTES + DDGSYNCCRYPTO_SECRET_KEY_SIZE], nonceBytes, crypto_secretbox_NONCEBYTES);
+
+    return DDGSYNCCRYPTO_OK;
+}
+
+int main(int argc, char **argv) {
+    if (argc != 4) {
+        printf("Usage: %s secretkey username password\n", argv[0]);
+        return 1;
+    }
+
+    unsigned char primaryKey[DDGSYNCCRYPTO_PRIMARY_KEY_SIZE];
+    unsigned char secretKey[DDGSYNCCRYPTO_SECRET_KEY_SIZE];
+    unsigned char protectedSecretKey[DDGSYNCCRYPTO_PROTECTED_SECRET_KEY_SIZE];
+    unsigned char passwordHash[DDGSYNCCRYPTO_HASH_SIZE];
+    
+    DDGSyncCryptoResult res;
+    {
+        int err = sodium_base642bin(secretKey, DDGSYNCCRYPTO_SECRET_KEY_SIZE, argv[1], strlen(argv[1]), NULL, NULL, NULL, sodium_base64_VARIANT_ORIGINAL);
+        if (err != 0) {
+            printf("err %d b64 decoding secretKey\n", err);
+            return 2;
+        }
+    }
+
+    res = ddgSyncGenerateAccountKeysWithSecretKey(primaryKey, secretKey, protectedSecretKey, passwordHash, argv[2], argv[3]);
+    if (res != 0) {
+        printf("errorcode: %d\n", res);
+        return 3;
+    }
+
+    {
+        unsigned char printbuf[sizeof(primaryKey)*2];
+        sodium_bin2base64((char*)printbuf, sizeof(printbuf), primaryKey, sizeof(primaryKey), sodium_base64_VARIANT_ORIGINAL);
+        printf("%s\n", printbuf);
+    }
+    {
+        unsigned char printbuf[sizeof(secretKey)*2];
+        sodium_bin2base64((char*)printbuf, sizeof(printbuf), secretKey, sizeof(secretKey), sodium_base64_VARIANT_ORIGINAL);
```

Comment:
Since you're passing the secret key in the input, there's no need to include it in the output. You could remove it, _if you want_.